csi.proto

```diff
@@ -271,15 +271,18 @@ message VolumeMetadata {
 }
 
 // A standard way to encode credential data. The total bytes of the values in
-// the Data field must be less than MaxSecretSize bytes.
+// the Data field must be less than 1 Mebibyte.
 message Credentials {
-  // Data contains the credential data, for example username and password. Each
-  // key must consist of alphanumeric characters, '-', '_' or '.'. The
-  // serialized form of the secret data is a base64 encoded string, representing
-  // the arbitrary (possibly non-string) data value here.
-  // This information is sensitive and should be treated as such (not logged,
-  // etc.)
-  map<string, bytes> Data = 1;
+  // Data contains the credential data, for example username and password.
+  // Each key must consist of alphanumeric characters, '-', '_' or '.'.
+  // Each value MUST contain a valid string. An SP MAY choose to accept binary
+  // (non-string) data by using a binary-to-text encoding scheme, like base64.
+  // An SP SHALL advertise the requirements for credentials in documentation.
+  // COs SHALL permit users to pass through the required credentials.
+  // This information is sensitive and MUST be treated as such (not logged,
+  // etc.) by the CO.
+  // This field is REQUIRED.
+  map<string, string> Data = 1;
```

Comment:
I'd prefer we use `map<string, bytes>` here and don't even mention base64 here. By default, the `bytes` type in protobuf will map to base64 encoded string in JSON representation (https://developers.google.com/protocol-buffers/docs/proto3#json).

Using `map<string, string>` here will force all COs to do base64 encoding to convert a binary secret to a string. If the SP expect the binary format anyway, it'll need to do a base64 decoding. I feel that this extra complexity is unnecessary given that protobuf already handles it.